src/couch/test/couch_changes_tests.erl

```diff
@@ -837,6 +843,27 @@ wait_finished({_, ConsumerRef}) ->
         ]})
     end.
 
+
+reset_row_notifications() ->
+    receive
+        row ->
+            reset_row_notifications()
+    after 0 ->
+        ok
+    end.
+
+
+wait_row_notifications(N) ->
+    receive
+        row when N == 1 ->
+            ok;
+        row when N > 1 ->
+            wait_row_notifications(N - 1)
+    after ?TIMEOUT ->
+        timeout
```

Comment:
I'd say `throw` here or raise an error, going to be less confusing than none-match crash in the test's logs in case of the actual timeout. pause and unpause do creative `?assertNotEqual(timeout, Resp),` for the same effect.